funcx_sdk/funcx/sdk/asynchronous/polling_task.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+from asyncio import AbstractEventLoop
+import time
+from funcx.sdk.asynchronous.funcx_task import FuncXTask
+
+
+class PollingTask:
```

Comment:
This class feels like a lot of boilerplate to do a fairly common thing. Is there not a well maintained async library or similar that has this code in it already? For example, similar to the retry library - very small, does one thing very well.